samples/interactive-tutorials/src/main/java/search/tutorials/search_with_boost_spec.md

```diff
@@ -0,0 +1,221 @@
+
+1. Find your service account on the [IAM page](https://console.cloud.google.com/iam-admin/iam),
+   click `Edit` icon and add the roles 'Storage Admin' and 'BigQuery Admin. It may take a while for the changes to take effect.
+
+1. Copy the service account email in the field Principal.
+
+### Set up authentication
+
+To run a code sample from the Cloud Shell, you need to be authenticated using the service account credentials.
+
+1. Login with your user credentials.
+    ```bash
+    gcloud auth login
+    ```
+
+1. Type `Y` and press **Enter**. Click the link in Terminal. A browser window should appear asking you to log in using your Gmail account.
+
+1. Provide the Google Auth Library with access to your credentials and paste the code from the browser to the Terminal.
+
+1. Upload your service account key JSON file and use it to activate the service account:
+
+    ```bash
+    gcloud iam service-accounts keys create ~/key.json --iam-account <YOUR_SERVICE_ACCOUNT_EMAIL>
+    ```
+
+    ```bash
+    gcloud auth activate-service-account --key-file ~/key.json
+    ```
+
+1. Set key as the GOOGLE_APPLICATION_CREDENTIALS environment variable to be used for requesting the Retail API:
+    ```bash
+    export GOOGLE_APPLICATION_CREDENTIALS=~/key.json
+    ```
+
+**Note**: Click the copy button on the side of the code box to paste the command in the Cloud Shell terminal and run it.
+
+### Set the PROJECT_NUMBER environment variable
+
+Because you are going to run the code samples in your own Google Cloud project, you should specify the **project_number** as an environment variable. It will be used in every request to the Retail API.
+
+1. You can find the ```project_number``` in the Project Info card displayed on **Home/Dashboard**.
+
+1. Set the environment variable with the following command:
+    ```bash
+    export PROJECT_NUMBER=<YOUR_PROJECT_NUMBER>
+    ```
+
+### Google Cloud Retail libraries
+
+You can find the Java Google Retail library is described
+[here](https://googleapis.dev/java/google-cloud-retail/latest/index.html)
+
+## Clone the Retail code samples
+
+This step is required if this is the first Retail API Tutorial you run.
+Otherwise, you can skip it.
+
+Clone the Git repository with all the code samples to learn the Retail features and check them in action.
+
+<!-- TODO(ianan): change the repository link -->
+1. Run the following command in the Terminal:
+    ```bash
+    git clone https://github.com/t-karasova/java-retail.git
+    ```
+
+The code samples for each of the Retail services are stored in different directories.
+
+2. Go to the `interactive-tutorials` directory. It's our starting point to run more commands.
+    ```bash
+    cd java-retail/samples/interactive-tutorials
+    ```
+
+## Boost by one criterion: condition and filtering expression
+
+The boosting specification looks like this:
+
+```
+  condition_boost_specs {
```

Comment:
I found only
String condition = "(colorFamilies: ANY(\"Blue\"))";
    float boost = 0.0f;

in the java file